src/dpl/src/Place.cpp

```diff
@@ -956,6 +1015,97 @@ PixelPt Opendp::binSearch(int x, const Cell* cell, int bin_x, int bin_y) const
   return PixelPt();
 }
 
+bool Opendp::checkRegionOverlap(const Cell* cell,
+                                int x,
+                                int y,
+                                int x_end,
+                                int y_end) const
+{
+  // TODO: Investigate the caching of this function
+  // it is called with the same cell and x,y,x_end,y_end multiple times
+  debugPrint(logger_,
+             DPL,
+             "detailed",
+             1,
+             "Checking region overlap for cell {} at x[{} {}] and y[{} {}]",
+             cell->name(),
+             x,
+             x_end,
+             y,
+             y_end);
+  auto row_info = getRowInfo(cell);
+  int min_row_height = row_height_;
+  bgBox queryBox(bgPoint(x * site_width_,
+                         map_coordinates(y, row_info.first, min_row_height)
+                             * min_row_height),
+                 bgPoint(x_end * site_width_ - 1,
+                         map_coordinates(y_end, row_info.first, min_row_height)
+                                 * min_row_height
+                             - 1));
+
+  std::vector<bgBox> result;
+  findOverlapInRtree(queryBox, result);
+
+  if (cell->region_
+      && result.size()
+             != 1) {  // cell has a region, so the overlap should only be with
+                      // one box (for that region) in the rtree
+    debugPrint(
+        logger_,
+        DPL,
+        "detailed",
+        1,
+        "Cell had a region, but it didn't overlap with it!!. Overlap size {}",
+        result.size());
+    for (const auto& box : result) {
+      debugPrint(logger_,
+                 utl::DPL,
+                 "detailed",
+                 1,
+                 "Overlap found between queryBox ({} , {}) - ({} , {}) and bbox"
+                 "({} , {}) - ({}, {})",
+                 queryBox.min_corner().x(),
+                 queryBox.min_corner().y(),
+                 queryBox.max_corner().x(),
+                 queryBox.max_corner().y(),
+                 box.min_corner().x(),
+                 box.min_corner().y(),
+                 box.max_corner().x(),
+                 box.max_corner().y());
```

Comment:
It would be nice to have stream operators so you can just write
 "Overlap found between queryBox {} and bbox {}", queryBox, box